File: schemaorggo/businessfunction.go
package schemaorggo

import "encoding/json"

// BusinessFunction see : https://schema.org/BusinessFunction
type BusinessFunction struct {
	Enumeration

	typeContext

	// SupersededBy see : http://meta.schema.org/supersededBy
	// Relates a term (i.e. a property, class or enumeration) to one that supersedes it.
	// types : Class Enumeration Property
	SupersededBy []interface{} `json:"supersededBy,omitempty"`
}

func (v BusinessFunction) IntoMap(intop *map[string]interface{}) error {
	if intop == nil {
		return nil
	}

	v.Enumeration.IntoMap(intop)

	into := *intop

	{
		var value interface{} = v.SupersededBy
		if len(v.SupersededBy) == 1 {
			value = v.SupersededBy[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["supersededBy"] = json.RawMessage(b)
		}
	}

	*intop = into

	return nil
}

func (v BusinessFunction) AsMap() (map[string]interface{}, error) {
	data := map[string]interface{}{}
	err := v.IntoMap(&data)
	if err != nil {
		return nil, err
	}

	data["@context"] = "http://schema.org"
	data["@type"] = "BusinessFunction"

	return data, nil
}

func (v BusinessFunction) MarshalJSON() ([]byte, error) {
	data, err := v.AsMap()
	if err != nil {
		return nil, err
	}

	return json.Marshal(data)
}

File: schemaorgphp/classes/business_function.php
<?php

class BusinessFunction extends Enumeration implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'BusinessFunction';
	
	/**
	 * Relates a term (i.e. a property, class or enumeration) to one that supersedes it.
	 * see : http://meta.schema.org/supersededBy
	 * @var \Class|\Class[]|\Enumeration|\Enumeration[]|\Property|\Property[]
	 */
	public var $superseded_by;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'BusinessFunction'
		);
		
		$serialized = so_json_serialize( $this->superseded_by );
		if ( ! empty( $serialized ) ) {
			$out['supersededBy'] = $serialized;
		}
		
		return $out;
	}
}
